netctl/netctl.go

```diff
@@ -401,3 +402,14 @@ func dumpList(ctx *cli.Context, list []map[string]interface{}) {
 	os.Stdout.Write(content)
 	os.Stdout.WriteString("\n")
 }
+
+func showVersion(ctx *cli.Context) {
```

Comment:
@mapuri - no, I am going to take it personally :-)

Jokes apart, I appreciate your comments! Let me see if we can conclude on the right balance that we need now.

> it doesn't address something similar for netplugin and k8s plugins.

If I am a user I'd use this command to find out things that I'd care about. Typically, user would care about the API version (to know what version an API is), Netmaster version (to know if certain features are available, etc.), and a client version (which can tell if the API that we document is supported by a given client version). Netplugin and k8s version are really for debugging, because they have no user implication. This is to say that if we have incompatible netplugin and netmaster version then we should go to lowest common denominator of API version, or disallow netplugin from them coming up and form some form of fully functional adjacency. What I thought I was doing was the user aspect of things - of course without introducing API version, because it requires a different set of changes. 

Having said that, showing netplugin version here, require `host` object that I have to wait for @abhinandanpb's changes for.

> it doesn't deal with multiple instances of netmaster itself

Could we do this as a part of doing active-active redundancy model for netmaster? 

> It is not useful for compatability as well since it doesn't talks about what API versions are being used by the client and server

Versions are to know what features are being used, compatibility is with API versioning. I felt we could address the compability with API versioning, while addressing the feature-support with this versioning

> we are adding a new endpoint now (ironically for bringing in versions), without addressing what happens when we decide to change contents of version.Info structure itself.

Ideally the first endpoint to be added is version, however this is needed before official 0.1 release, hence this PR. About changing the Version Info structure, we deal with the same way we modify the rest. Provide with API version in the version object so a client can know what language to talk to with here.

This discussion and thoughts will become handy regardless of whether we make the changes now or later. 

Thoughts?